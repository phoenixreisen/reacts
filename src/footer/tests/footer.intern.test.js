/**
 * @jest-environment jsdom
 */
 import { render, fireEvent, waitFor, screen } from '@testing-library/react';
 import renderer from 'react-test-renderer';
 import Footer from '../footer.intern.r';
 import React from 'react';

 describe('footer should', () => {

    test('render correctly when logged out', () => {
        const component = renderer.create(
            <Footer loggedIn={false} />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('render correctly when logged in', () => {
        const component = renderer.create(
            <Footer
                loggedIn={true}
                username='Fabian'
            />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });
 });