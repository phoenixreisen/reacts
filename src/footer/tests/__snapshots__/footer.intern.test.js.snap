// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer should render correctly when logged in 1`] = `
<article
  className="intern-footer"
>
  <div
    className="wrapper wrapper--large"
  >
    <hr />
    <div
      className="tr mr3"
    >
      <span>
        eingeloggt als 
        <span
          className="f4"
        >
          Fabian
        </span>
      </span>
      <span
        className="mh3"
      >
        |
      </span>
      <a
        href="/cdn/mitarbeiter/login/"
      >
        <i
          className="fas fa-sign-out-alt mr1"
        />
         abmelden
      </a>
    </div>
  </div>
</article>
`;

exports[`footer should render correctly when logged out 1`] = `
<article
  className="intern-footer"
>
  <div
    className="wrapper wrapper--large"
  >
    <hr />
    <div
      className="tr mr3"
    >
      <a
        href="/cdn/mitarbeiter/login/"
      >
        <i
          className="fas fa-sign-in-alt mr1"
        />
         anmelden
      </a>
    </div>
  </div>
</article>
`;
